### PR TITLE
fix(websocket): don't dial `/dnsaddr` addresses

### LINKED ISSUE #5601: `/dnsaddr` is handled as `/dns4` in `libp2p-websocket` and `libp2p-websocket-websys`
### Summary

`/dnsaddr` is handled as `/dns4` in `libp2p-websocket` and `libp2p-websocket-websys`. This is incorrect because `/dnsaddr` is only for quering `_dnsaddr.domain` from DNS.

`/dnsaddr` support in WebSocket is tracked in https://github.com/libp2p/rust-libp2p/issues/5529.

libp2p-websocket:
https://github.com/libp2p/rust-libp2p/blob/a2a281609a0a64b211f7917aa856924983b63200/transports/websocket/src/framed.rs#L511-L516

libp2p-websocket-websys:
https://github.com/libp2p/rust-libp2p/blob/a2a281609a0a64b211f7917aa856924983b63200/transports/websocket-websys/src/lib.rs#L131-L136

### Expected behavior

`TransportError::MultiaddrNotSupported` should be returned from `dial`.

### Actual behavior

`/dnsaddr` multiaddress is dialed as if it was `/dns4`.

### Relevant log output

_No response_

### Possible Solution

_No response_

### Version

0.54.1

### Would you like to work on fixing this bug ?

Maybe

### PR DESCRIPTION
**Pull request by hmzdot**
## Description

Returns `Error::InvalidMultiaddr` when `parse_ws_dial_addr` is called with `/dnsaddr`.

As per its specification, `/dnsaddr` domains are not meant to be directly dialed, instead it should be appended with `_dnsaddr.` and used for DNS lookups afterwards

Related: #5529
Fixes: #5601 

## Notes & open questions

* Is it okay to return an error, or should I perform a DNS lookup and resolve that DNS afterwards if address has `/dnsaddr`?
* If so, how should I handle that case where DNS lookup returns multiple multiaddrs?

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -151,4 +151,8 @@ clippy.manual_let_else = "warn"
 clippy.dbg_macro = "warn"
 
 [workspace.metadata.release]
-pre-release-hook = ["/bin/sh", '-c', '/bin/sh $WORKSPACE_ROOT/scripts/add-changelog-header.sh'] # Nested use of shell to expand variables.
+pre-release-hook = [
+    "/bin/sh",
+    '-c',
+    '/bin/sh $WORKSPACE_ROOT/scripts/add-changelog-header.sh',
+] # Nested use of shell to expand variables.
```
Comment (dariusc93) [inline Cargo.toml]:
```suggestion
pre-release-hook = ["/bin/sh", '-c', '/bin/sh $WORKSPACE_ROOT/scripts/add-changelog-header.sh'] # Nested use of shell to expand variables.
```

**Review by dariusc93 — APPROVED** [review]
LGTM.

File: Cargo.toml (hunk 1 above)
Comment (dariusc93) [reply Cargo.toml]: This just reverts back to the previous format is all :)

**Comment by oblique** [comment]
[websocket-websys](https://github.com/libp2p/rust-libp2p/blob/8ceadaac5aec4b462463ef4082d6af577a3158b1/transports/websocket-websys/src/lib.rs#L134) needs this change too.

**Comment by hmzdot** [comment]
> [bsys](https://github.com/libp2p/rust-libp2p/blob/8ceadaac5aec4b462463ef4082d6af577a3158b1/transports/websocket-websys/src/lib.rs#L134) needs this change too.

I see, will update there as well

**Comment by DougAnderson444** [comment]
Hello

I just opened https://github.com/libp2p/rust-libp2p/issues/5619 because I want to dial `/dnsaddr` from [webrtc-websys](https://github.com/libp2p/rust-libp2p/tree/master/transports/webrtc-websys). Then I noticed this issue.

Would it make sense to simply support DNS multiaddr resolution in the browser?

**Comment by DougAnderson444** [comment]
Also https://github.com/libp2p/rust-libp2p/issues/5531 could enable this?

**Comment by oblique** [comment]
@DougAnderson444 As described in #5601, this PR removes an incorrect handling of `/dnsaddr`. #5531 is the only way forward which needs to implement resolution of only `/dnsaddr` multiaddresses through a new `Transport` that targets only browsers.

**Comment by hmzdot** [comment]
Updated websocket-websys (bf23e92b6ccd60ebe65d08d2436dee62d80946ba) so that it returns `None` when a `/dnsaddr` is extracted.
This should be enough to fix incorrect handling of `/dnsaddr` and to resolve them correctly we can continue from #5531

cc: @dariusc93

**Review by oblique — APPROVED** [review]

**Comment by hmzdot** [comment]
Previous CI runs failed because I didn't update top-level versions, my bad. Can you restart the workflow? @oblique

**Comment by oblique** [comment]
cc @jxs

**Review by jxs — APPROVED** [review]
LGTM Thanks Yianis!

**Pull request merged**